base/operators.jl

```diff
@@ -153,10 +153,15 @@ const ÷ = div
 # mod returns in [0,y) or (y,0] (for negative y),
 # whereas mod1 returns in (0,y] or [y,0)
 mod1{T<:Real}(x::T, y::T) = (m=mod(x,y); ifelse(m==0, y, m))
+# mod with offset n
+mod{T<:Real}(x::T, y::T, n::T) = mod(x+y-n,y)+n
```

Comment:
If `n == 0` then is the term `mod(x + y, y`) which has the same result as `mod(x, y)`. I will also include tests to check if everything works fine for negative `n`.

In case of the `x+y-n` term, I followed the conventions of the `mod1` function. It may faster overflow when `y` and `n` are very large.